tests/components/datadog/test_init.py

```diff
@@ -10,59 +9,47 @@
     STATE_ON,
 )
 import homeassistant.core as ha
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
 
-from tests.common import assert_setup_component, get_test_home_assistant
+from tests.async_mock import MagicMock, patch
+from tests.common import assert_setup_component
 
 
-class TestDatadog(unittest.TestCase):
-    """Test the Datadog component."""
+async def test_invalid_config(hass):
+    """Test invalid configuration."""
+    with assert_setup_component(0):
+        assert not await async_setup_component(
+            hass, datadog.DOMAIN, {datadog.DOMAIN: {"host1": "host1"}}
+        )
 
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.tear_down_cleanup)
 
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+async def test_datadog_setup_full(hass):
+    """Test setup with all data."""
+    config = {datadog.DOMAIN: {"host": "host", "port": 123, "rate": 1, "prefix": "foo"}}
+    hass.bus.listen = MagicMock()
 
-    def test_invalid_config(self):
-        """Test invalid configuration."""
-        with assert_setup_component(0):
-            assert not setup_component(
-                self.hass, datadog.DOMAIN, {datadog.DOMAIN: {"host1": "host1"}}
-            )
+    with patch("homeassistant.components.datadog.initialize") as mock_init, patch(
+        "homeassistant.components.datadog.statsd"
+    ):
+        assert await async_setup_component(hass, datadog.DOMAIN, config)
 
-    @mock.patch("homeassistant.components.datadog.statsd")
-    @mock.patch("homeassistant.components.datadog.initialize")
-    def test_datadog_setup_full(self, mock_connection, mock_client):
-        """Test setup with all data."""
-        self.hass.bus.listen = mock.MagicMock()
+        assert mock_init.call_count == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)